iosapp/app/Lib/Animation.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+import SwiftUI
+
+func delayed(by delay: Duration, _ callback: @escaping () -> Void) {
+  DispatchQueue.main.asyncAfter(deadline: .now() + delay.inMilliseconds / 1000) {
+    callback()
+  }
+}
+
+extension Duration {
+  var inMilliseconds: Double {
+    let v = components
+    return Double(v.seconds) * 1000 + Double(v.attoseconds) * 1e-15
+  }
+}
+
+struct Vector: Equatable, Identifiable, AdditiveArithmetic {
+  var x: Double
+  var y: Double
+
+  var magnitude: Double {
+    sqrt(self.x * self.x + self.y * self.y)
+  }
+
+  var isOrigin: Bool {
+    self.x == 0 && self.y == 0
+  }
+
+  var normalized: Vector {
+    self.scaledBy(1 / self.magnitude)
+  }
+
+  func scaledBy(_ scalar: Double) -> Vector {
+    .init(x: self.x * scalar, y: self.y * scalar)
+  }
+
+  static var zero: Vector {
+    .init(x: 0, y: 0)
+  }
+
+  static func - (lhs: Vector, rhs: Vector) -> Vector {
+    .init(x: lhs.x - rhs.x, y: lhs.y - rhs.y)
+  }
+
+  static func + (lhs: Vector, rhs: Vector) -> Vector {
+    .init(x: lhs.x + rhs.x, y: lhs.y + rhs.y)
+  }
+
+  static func += (lhs: inout Vector, rhs: Vector) {
+    lhs = lhs + rhs
+  }
+
+  var id: UUID
```

Comment:
just a nitpick, but i generally like to keep stored properties together, up near the top, so i can grok at a glance what  the struct actually contains in memory.